cpp/examples/hybrid_scan_io/common_utils.cpp

```diff
@@ -91,20 +95,35 @@ void check_tables_equal(cudf::table_view const& lhs_table,
   }
 }
 
-cudf::host_span<uint8_t const> fetch_footer_bytes(cudf::host_span<uint8_t const> buffer)
+io_backend::io_backend(cudf::host_span<std::byte const> buffer, rmm::cuda_stream_view stream)
+  : _host_buffer_source(std::make_unique<host_buffer_source>(buffer)), _stream(stream)
+{
+  _datasource = cudf::io::datasource::create(_host_buffer_source.get());
```

Comment:
Nit: `_datasource` can be constructed in the initializer list, after `_host_buffer_source`.